### PR TITLE
Add ESP32 support

### PR DESCRIPTION
**Pull request by Acuadros95**

### CODE REVIEW

File: src/micro_ros_arduino.h [hunk 1]
```diff
@@ -102,10 +102,13 @@ struct micro_ros_agent_locator {
 };
 
 static inline void set_microros_wifi_transports(char * ssid, char * pass, char * agent_ip, uint agent_port){
-	while (WiFi.begin(ssid, pass) != WL_CONNECTED)
-	{
-    	delay(500);
-	}
+
```
Comment (pablogs9) [inline src/micro_ros_arduino.h:105]: Is this checked in other boards?

Comment (Acuadros95) [reply src/micro_ros_arduino.h:105]: Yes, fixed a bug here: https://github.com/micro-ROS/micro_ros_arduino/pull/680/commits/de6bc5bea0d6cff7fd7f9d7fe3446cd687aaaa55

File: src/micro_ros_arduino.h [hunk 2]
```diff
@@ -102,10 +102,12 @@ struct micro_ros_agent_locator {
...
+	WiFi.begin(ssid, pass);
+
+    while (WiFi.status() != WL_CONNECTED) {
```
Comment (pablogs9) [inline src/micro_ros_arduino.h:108]: WHy aren't we trying in a loop to connect using `begin` instead of trying once and checking the status?

Comment (Acuadros95) [reply src/micro_ros_arduino.h:108]: `begin` configures the network `ssid` and `pass` and checks the connection using the `status`, as you can see here: https://github.com/arduino-libraries/WiFi/blob/master/src/WiFi.cpp#L98

There is no need to reconfigure this values again, when we can just check the status of the connection.

Comment (pablogs9) [reply src/micro_ros_arduino.h:108]: Nice, thanks for the clarification

File: extras/library_generation/library_generation.sh [hunk 3]
```diff
@@ -53,171 +53,117 @@ pushd firmware/mcu_ws > /dev/null
 
 popd > /dev/null
 
+# Workaround. Remove when https://github.com/ros2/rosidl/pull/596 is merged
+touch firmware/mcu_ws/ros2/common_interfaces/actionlib_msgs/COLCON_IGNORE;
+touch firmware/mcu_ws/ros2/common_interfaces/std_srvs/COLCON_IGNORE;
+touch firmware/mcu_ws/ros2/example_interfaces/COLCON_IGNORE;
```
[the remark this answers is not in the document]
Comment (Acuadros95) [reply extras/library_generation/library_generation.sh]: Yes! Deleted here: https://github.com/micro-ROS/micro_ros_arduino/pull/680/commits/cd7c3d24a55b438e3bb361ed79457fdf86cfb044

**Comment by Acuadros95** [comment]
@mergify backport foxy main

File: extras/library_generation/library_generation.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (Acuadros95) [reply extras/library_generation/library_generation.sh]: This includes new packages and modifies all libraries, so I think its better to rebuild.
This commit will trigger the build and will serve as a test: https://github.com/micro-ROS/micro_ros_arduino/pull/680/commits/89222760c86a985b17e6c23918e9e0574db24238

**Review by pablogs9 — APPROVED** [review]

**Pull request merged**